Runigram.java

```diff
@@ -41,8 +55,12 @@ public static Color[][] read(String fileName) {
 		// For each pixel (i,j), reads 3 values from the file,
 		// creates from the 3 colors a new Color object, and 
 		// makes pixel (i,j) refer to that object.
-		//// Replace the following statement with your code.
-		return null;
+		for (int row = 0; row < image.length; row++) {
+			for (int col = 0; col < image[0].length; col++) {
+				image[row][col] = new Color(in.readInt(), in.readInt(), in.readInt());

```

Comment:
Better to be clear what values you're expecting for.
```suggestion
				int red = in.readInt();
				int blue = in.readInt();
				int green = in.readInt();
				image[row][col] = new Color(red, blue, green);
```